Mailer/app/models/Repositories/UserSubscriptionsRepository.php

```diff
@@ -149,4 +159,13 @@ public function getAllSubscribersDataForMailTypes(array $mailTypeIds)
             ->where('mail_type_id', $mailTypeIds)
             ->group('mail_type_id, subscribed');
     }
+
+    public function unsubscribeUserVariant(ActiveRow $mailType, $userId, $variantId, $email, $utmParams = [], $emitEvent = true)
+    {
+        $userSubscription = $this->getTable()->where(['user_id' => $userId, 'mail_type_id' => $mailType->id])->limit(1)->fetch();
```

Comment:
You need to check if `$userSubscription` actually exists. Otherwise `removeSubscribedVariant` can end up with error.  
_(please test regulard and also edge cases before submitting PR)_